File: myScala1/src/main/scala/chapter03/Test01.scala
package chapter03

/**
 * 比较运算符
 * scala 中 == 字符串的 == 相当于equals
 * 如果想判断引用内存地址是不是相同,用eq方法
 */
object Test01 {

  def main(args: Array[String]): Unit = {
    val s1:String  = "hello"
    val s2:String  = new String("hello")

//    println(s1 == s2)
//
//    println(s1.equals(s2))
//    println(s1.eq(s2))

    // 判断是否为空
    def isNotEmpty(str:String):Boolean = {
      return !(str == "")
    }

    val a = null
    println(isNotEmpty(a))
  }

}

File: myScala1/src/main/java/jchapter03/Test01.java
package jchapter03;

public class Test01 {
    public static void main(String[] args) {
        String s1  = "hello";
        String s2  = new String("hello");

        System.out.println(s1 == s2);
        System.out.println(s1.equals(s2));

        Long a = 100L;
        a += 1;
        System.out.println(a);

        System.out.println(++a);
        System.out.println(a);
    }
}
